### ISSUE TITLE
Evaluation function for datasets mentioned in the paper

### ISSUE THREAD

**Issue by paulzyzy** [LiaoMengqi/HMoRA#2]
Hi, thanks for providing this wonderful work! 
I am new to language models and I am wondering if you could provide the evaluation functions for each datasets you mentioned in the paper.
I am not quite sure how should I format each dataset to do the test and validation so it would be very kind if you could help me figure this out.
Thanks again for your work and patience.

**Comment by paulzyzy** [comment]
And also I just found out that the test set from both swag and commensenseqa dont have label/answerkey, but according to the paper, you sampled your test set from original test set of those two datasets. I am wondering if you could share some details of how you did the evaluation without label/answerkey?
Thanks again for your time and help!

**Comment by LiaoMengqi** [comment]
We appreciate your feedback. We formatted multiple-choice questions in a specific structure and selected the token with the highest logits as the model’s answer. The latest version of the code has already been updated. Admittedly, this evaluation method is only applicable to multiple-choice questions and does not support the use of Chain-of-Thought (CoT) reasoning. A more recommended approach is to prompt the model to reason first and then to provide its final answer within a \boxed{}  We then match the answer contained within the  \boxed{} . For the SWAG and CommonsenseQA datasets, our test set was split from the validation set, as noted in the appendix.
